pkg/clusteroperator/clusteroperator.go

```diff
@@ -0,0 +1,233 @@
+package clusteroperator
+
+import (
+	"context"
+	"time"
+
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "github.com/openshift/api/config/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	fiveScds time.Duration = 5 * time.Second
+	isTrue                 = "True"
+)
+
+// ClusterOperatorBuilder provides struct for ClusterOperator object.
+type ClusterOperatorBuilder struct {
+	// ClusterOperator definition. Used to create a clusteroperator object.
+	Definition *v1.ClusterOperator
+	// Created clusteroperator object.
+	Object *v1.ClusterOperator
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+	// Used in functions that define or mutate ClusterOperator definition. errorMsg is processed before the
+	// ClusterOperator object is created.
+	errorMsg string
+}
+
+// Pull pulls existing ClusterOperator from cluster.
+func Pull(apiClient *clients.Settings, name string) (*ClusterOperatorBuilder, error) {
+	glog.V(100).Infof("Pulling existing ClusterOperator object name %s from cluster", name)
+
+	builder := ClusterOperatorBuilder{
+		apiClient: apiClient,
+		Definition: &v1.ClusterOperator{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the ClusterOperator is empty")
+
+		builder.errorMsg = "ClusterOperator 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("ClusterOperator object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Exists checks whether the given ClusterOperator exists.
+func (builder *ClusterOperatorBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if ClusterOperator %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Pull()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Pull fetches the defined ClusterOperator from the cluster.
+func (builder *ClusterOperatorBuilder) Pull() (*v1.ClusterOperator, error) {
```

Comment:
Please re-name it to Get. 
We are planning to implement common interface struct soon for all objects under the project.

Reply:
done